src/cascadia/TerminalSettingsEditor/ActionsViewModel.idl

```diff
@@ -1,60 +1,188 @@
-        String AcceptButtonName { get; };
-        String DeleteButtonName { get; };
-        Windows.UI.Xaml.Media.Brush ContainerBackground { get; };
-
-        void EnterHoverMode();
-        void ExitHoverMode();
-        void ActionGotFocus();
-        void ActionLostFocus();
-        void EditButtonGettingFocus();
-        void EditButtonLosingFocus();
-        IObservableVector<String> AvailableActions { get; };
+        Boolean IsInEditMode { get; };
         void ToggleEditMode();
         void AttemptAcceptChanges();
         void CancelChanges();
-        void DeleteKeyBinding();
+        void DeleteKeyChord();
+        String CancelButtonName { get; };
+        String AcceptButtonName { get; };
+        String DeleteButtonName { get; };
 
-        event Windows.Foundation.TypedEventHandler<KeyBindingViewModel, ModifyKeyBindingEventArgs> ModifyKeyBindingRequested;
-        event Windows.Foundation.TypedEventHandler<KeyBindingViewModel, Microsoft.Terminal.Control.KeyChord> DeleteKeyBindingRequested;
+        event Windows.Foundation.TypedEventHandler<KeyChordViewModel, Microsoft.Terminal.Control.KeyChord> AddKeyChordRequested;
+        event Windows.Foundation.TypedEventHandler<KeyChordViewModel, ModifyKeyChordEventArgs> ModifyKeyChordRequested;
+        event Windows.Foundation.TypedEventHandler<KeyChordViewModel, Microsoft.Terminal.Control.KeyChord> DeleteKeyChordRequested;
     }
 
+    enum ActionsSubPage
+    {
+        Base = 0,
+        Edit = 1
+    };
+
     runtimeclass ActionsViewModel : Windows.UI.Xaml.Data.INotifyPropertyChanged
     {
         ActionsViewModel(Microsoft.Terminal.Settings.Model.CascadiaSettings settings);
+        void UpdateSettings(Microsoft.Terminal.Settings.Model.CascadiaSettings settings);
 
-        void OnAutomationPeerAttached();
-        void AddNewKeybinding();
+        void AddNewCommand();
 
-        IObservableVector<KeyBindingViewModel> KeyBindingList { get; };
-        event Windows.Foundation.TypedEventHandler<Object, Object> FocusContainer;
-        event Windows.Foundation.TypedEventHandler<Object, Object> UpdateBackground;
+        ActionsSubPage CurrentPage;
+        CommandViewModel CurrentCommand;
+        Boolean DisplayBadge { get; };
+
+        void AttemptAddOrModifyKeyChord(KeyChordViewModel senderVM, String commandID, Microsoft.Terminal.Control.KeyChord newKeys, Microsoft.Terminal.Control.KeyChord oldKeys);
+        void AttemptDeleteKeyChord(Microsoft.Terminal.Control.KeyChord keys);
+        void AttemptAddCopiedCommand(Microsoft.Terminal.Settings.Model.Command newCommand);
+        void AttemptRegenerateCommandID(Microsoft.Terminal.Settings.Model.Command command);
+
+        IObservableVector<CommandViewModel> CommandList { get; };
+        void CmdListItemClicked(IInspectable sender, Windows.UI.Xaml.Controls.ItemClickEventArgs args);
+
+        Windows.Foundation.Collections.IMap<Microsoft.Terminal.Settings.Model.ShortcutAction, String> AvailableShortcutActionsAndNames();
```

Comment:
Careful, these two functions let you break out of "viewmodel" land and get into "real settings object" land (!)

Do they need to be part of the public API surface for ActionsViewModel?